sdk/program/src/vote/state/vote_state_deserialize.rs

```diff
@@ -86,21 +86,12 @@ fn read_prior_voters_into<T: AsRef<[u8]>>(
     if !is_empty {
```

Comment:
if is_empty=true _and_ prior_voters.buf is filled, we diverge from bincode
i think? One could maliciously create such a vote state.

(I think our Arbitrary impl for CircBuf should probably additionally generate this)

Reply:
its true that it would diverge. how could one create such a vote state tho? i wrote this based on the fact that it seemed there was no code path to do that, all changes to circbuf go through the insert function (hence also why the `Arbitrary` impl works that way)

Reply:
I’m a hacker, I see this commit get deployed to mainnet, as soon as it is I craft a special payload to split the cluster

Reply:
i was gonna say some stuff about how such a payload cant be delivered because it would fail vote program ownership, but it turns out theres negligible perf benefit to the special logic anyway. so, onwards!

Reply:
In the call site where I'm looking to optimize this (Bank::update_stakes_cache), I'm definitely seeing votes fail to deserialize, so surely this must be possible somehow?

Reply:
not sure what itd be withput seeing an example. in any event, should be the same as bincode now